### PR TITLE
[UI]: add a version tag to CatalogCard component

### LINKED ISSUE #688: Centralizing the CatalogCard component in sistent
#### Current Behavior

The `meshery-cloud` is still using its own `CatalogCard`, [this](https://github.com/layer5io/meshery-cloud/blob/master/ui/components/general/catalog-card/index.js#L95) needs to be centralized to sistent to consolidate these components.

#### Desired Behavior

Enhance the Sistent component to avoid creation and usage of a separate `CatalogCard` in `meshery-cloud` 

#### Implementation

- [ ] Adding version property to CatalogCardProps
```
type CatalogCardProps = {
  // ... existing props
  version?: string;
};
```
- [ ] Adding new styled component for the version tag
```
export const VersionTag = styled('div')(({ theme }) => ({
  position: 'absolute',
  top: '8px',
  right: '8px',
  backgroundColor: based on theme,
  color: based on theme,
  padding: '2px 6px',
  borderRadius: '4px',
  fontSize: '0.75rem',
  fontWeight: 'bold',
}));
```
- [ ] Updating the CatalogCard component to include the version tag
```
const CatalogCard: React.FC<CatalogCardProps> = ({
  pattern,
  patternType,
  cardHeight,
  cardWidth,
  cardStyles,
  cardLink,
  version,
}) => {
//....existing code 
  <ClassWrap catalogClassName={pattern?.catalog_data?.content_class} />
   {version && <VersionTag>v{version}</VersionTag>}
   <DesignType>{patternType}</DesignType>
```
If the CatalogCard is used in a component that  has a version it can then be passed like this:
```
<CatalogCard
  key={index}
  cardLink={`https://meshery.layer5.io/catalog/content/catalog/${item?.id}`}
  cardHeight="20rem"
  cardWidth="16rem"
  pattern={item}
  type="Catalog"
  patternType={item?.catalog_data?.type}
  version={item?.version} // Add this line if version tag is required
/>
```
Note: The above implementation is with reference to `CatalogCard` usage in [CatalogGrid](https://github.com/layer5io/layer5/blob/6606c3c53cdad37b80f9e3c02aa67c973b7db11c/src/sections/Meshery/Meshery-integrations/Individual-Integrations/CatalogGrid.js#L8) component of Layer5

#### Mockups

---

**[Contributor Guide](https://docs.meshery.io/project/contributing)**

- 🎨 Wireframes and [designs for Sistent site](https://www.figma.com/file/ekm0Hh6M25bSKpP5BcIJom/Sistent-Design-System?type=design&node-id=596%3A5227&mode=design&t=8MExPykHN1vQFCTZ-1) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)

### PR DESCRIPTION
**Pull request by SAHU-01**
**Notes for Reviewers**

This PR fixes #688

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

### CODE REVIEW

File: src/custom/CatalogCard/CatalogCard.tsx [hunk 1]
```diff
@@ -38,8 +38,21 @@ type CatalogCardProps = {
...
   cardStyles: React.CSSProperties;
   type: string;
+  version?: string;
 };
 
+export const VersionTag = styled('div')(({ theme }) => ({
+  position: 'absolute',
+  top: '40px',
+  right: '8px',
+  backgroundColor: theme.palette.primary.main,
```
Comment (sudhanshutech) [inline src/custom/CatalogCard/CatalogCard.tsx:48]: use color from palette @SAHU-01

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

File: src/custom/CatalogCard/CatalogCard.tsx [hunk 2]
```diff
@@ -38,8 +38,21 @@ type CatalogCardProps = {
...
   cardStyles: React.CSSProperties;
   type: string;
+  version?: string;
 };
 
+export const VersionTag = styled('div')(({ theme }) => ({
+  position: 'absolute',
+  bottom: '65px',
+  left: '18px',
+  backgroundColor: theme.palette.grey[500],
```
Comment (sudhanshutech) [inline src/custom/CatalogCard/CatalogCard.tsx:48]: not these, there is tokens we use you will get those in palette.js file.

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

File: src/custom/CatalogCard/CatalogCard.tsx (hunk 2 above)
Comment (SAHU-01) [reply src/custom/CatalogCard/CatalogCard.tsx:48]: > not these, there is tokens we use you will get those in palette.js file.

Thank you for sharing this, I'll make sure to take a look at it and modify this.

**Comment by sudhanshutech** [comment]
@dottharun

**Comment by SAHU-01** [comment]
@sudhanshutech based on the Wednesday call feedback that I received,  I need to add the flipable thingy and also centralize the catalog card and add on flip all the details that were available on meshery cloud. I'll be done with this over the weekend and will make sure to notify you.

**Comment by sudhanshutech** [comment]
oh its good. I was tagging @dottharun to confirm is he has some things to confirm here?

**Comment by dottharun** [comment]
@SAHU-01 Does this support dark mode, and can you provide screenshots if you can?

**Comment by SAHU-01** [comment]
@dottharun the tag color doesn't change with dark or light mode switch as of now if you want me to enable that, do let me know
image
image

**Comment by dottharun** [comment]
@SAHU-01 Yes, please use the same shades as the footer.

**Comment by SAHU-01** [comment]
@dottharun I did add the version background based on the color palette used for metrics container and it looks something like this:
for dark theme:
image
for light theme:
image

**Comment by SAHU-01** [comment]
@sudhanshutech, I have a couple of questions here:

1. should I move all styled components like this to `src>custom>CatalogCard>style.css`:
```
const BackFaceContent = styled(Box)(({ theme }) => ({
  position: 'absolute',
  background: `linear-gradient(to bottom right, black 40%, ${theme.palette.background.brand?.default})`,
  width: '100%',
  top: 0,
  left: 0,
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'left',
  padding: '16px',
  boxShadow: `2px 2px 3px 0px black`,
  borderRadius: '1rem'
}));
```
2. As discussed earlier for lining issues I've added a new interface do let em know if that requires changes.
3. @dottharun had suggested adding version background based on the color palette used for `metrics container` please take a look at it and let me know if that suits or we could opt for a different color palette
4. The linear gradient is not similar to the one on [Catalog Page](https://meshery.layer5.io/catalog) I was unable to find a figma for the flip catalog card, do let me know if that needs to be changed, I would need the exact hex codes though.

**Comment by SAHU-01** [comment]
Please find the demo screen here:

https://github.com/user-attachments/assets/c8849249-b816-4044-b9ea-2906b296187f

**Comment by sudhanshutech** [comment]
@SAHU-01 yes the styles should be in separate style folder

**Comment by dottharun** [comment]
@SAHU-01 If we are using the flip card and showing the version only on the backside, just use the same shades as in https://meshery.layer5.io/.

**Comment by sudhanshutech** [comment]
any more help needed @SAHU-01 ?

**Comment by SAHU-01** [comment]
@sudhanshutech yes the figma design? Does the gradient and padding look good to you in the recording? If yes I'll push the style file and tag theme changes now?

**Comment by sudhanshutech** [comment]
the color of metrics in from of card could be shade of white for dark mode. See which looks good from using the token.

**Comment by sudhanshutech** [comment]
@SAHU-01 have gone through the card here in cloud,: https://meshery.layer5.io/catalog?

**Comment by sudhanshutech** [comment]
@dottharun @vishalvivekm @captain-Akshay do we have figma design for this card?

**Comment by SAHU-01** [comment]
@dottharun, I've removed the flip logic from this PR and I've added the `versionTag` and the `versionTagPosition` props with proper styling as per your suggestion.
@sudhanshutech I'll be sending the other PR with centralized sistent logic soon.
Currently this is how it looks:

https://github.com/user-attachments/assets/3af4314b-855a-4d43-8b98-34d6e4750eeb

**Pull request closed without merging**